tiledb/sm/array_schema/dimension.h

```diff
@@ -1043,13 +1060,15 @@ class Dimension {
           "Domain check failed; Upper domain bound should "
           "not be smaller than the lower one"));
 
+    typedef typename std::make_unsigned<T>::type unsigned_t;
+
     // Domain range must not exceed the maximum uint64_t number
     // for integer domains
-    if ((uint64_t)(domain[1] - domain[0]) ==
-        std::numeric_limits<uint64_t>::max())
+    if ((unsigned_t)(domain[1] - domain[0]) ==
+        std::numeric_limits<unsigned_t>::max())
```

Comment:
There's only one pair of numbers where this condition is true, when the lower bound contains (signed) `min` and the upper bound contains `max`. Checking equality twice avoids relying on signed integer overflow, which is UB.